mobile/src/main/java/ch/epfl/sdp/mobile/application/tournaments/StoreDocumentTournament.kt

```diff
@@ -134,11 +114,83 @@ class StoreDocumentTournament(
               .whereEquals("roundDepth", round)
               .get<ChessDocument>()
               .toPoolResults()
-      createMatchesForRankedPlayers(results) { (round - 1).takeIf { it >= 1 } }
+      store.transaction { createMatchesForPoolResults(results) { (round - 1).takeIf { it >= 1 } } }
+    }
+  }
+
+  private fun Transaction<DocumentReference>.createPools(
+      tournamentDocumentReference: DocumentReference,
+      current: TournamentDocument,
+      maxPlayers: Int,
+      poolSize: Int,
+      bestOf: Int,
+      adminId: String,
+      players: List<String>
+  ) {
+    val chosenPlayers = players.shuffled().take(maxPlayers)
+    val pools =
+        chosenPlayers.chunked(poolSize).map { ids ->
+          ids.map { uid ->
+            uid to (get<ProfileDocument>(store.collection("users").document(uid))?.name ?: "")
+          }
+        }
+    val minOpponentsPerPool = pools.lastOrNull()?.size ?: 0
+
+    set(tournamentDocumentReference, current.copy(stage = StagePools))
+
+    pools
+        .asSequence()
+        .mapIndexed { index, participants ->
+          PoolDocument(
+              name = tournamentDetailsPoolName(index + 1),
+              tournamentId = reference.uid,
+              minOpponentsForAnyPool = minOpponentsPerPool,
+              remainingBestOfCount = bestOf,
+              tournamentBestOf = bestOf,
+              tournamentAdminId = adminId,
+              playerIds = participants.map { it.first },
+              playerNames = participants.map { it.second },
+          )
+        }
+        .forEach { set(store.collection(PoolDocument.Collection).document(), it) }
+  }
+
+  private fun Transaction<DocumentReference>.createFinalsMatchesForPlayers(
```

Comment:
Missing documentation :


```kotlin
/**
 * Creates the matches corresponding to the given finals phase.
 *
 * @receiver the ongoing [Transaction].
 * @param players the identifiers of the players, 
 * @param nextDepth the function which computes the depth of the elimination tree.
 */
private fun Transaction<DocumentReference>.createFinalsMatchesForPlayers()
```

Reply:
Fixed in https://github.com/epfl-SDP/android/pull/392/commits/a8685cd7be6104b3a4f8315aabff2c43f026a76b